src/components/Upload.jsx

```diff
@@ -39,6 +39,13 @@ const Upload = ({ setCurrentScript, titles, setTitles }) => {
         return response.json();
       })
       .then((title) => {
+        if (title === 'Could not find a title, potentially invalid file type') {
+          throw new Error(title);
+        }
+        // above potential error also has a 500 status so we check this error afterwards
+        if (response.status === 500) {
```

Comment:
I don't think it's possible for a 500 error to be caught here. It _should_ go directly to the .catch() block below